Symbolic Evauation

atondwal opened the pull request:
This patchset brings in the symbolic evaluation functionality from
LiquidHaskell into fixpoint. Still need to write plaintext fq parsing and
tests.

atondwal committed:
Moved Instantiate into fixpoint

atondwal committed:
Put instantiateAxioms in fp pipeline

It's now the first thing we do inside solveNative'

atondwal committed:
Removed `Show` constraint on instantiateAxiom

Needed to put instantiateAxiom inside solveNative

atondwal committed:
Rename Simplify into Rewrite and move getEqBody

ranjitjhala commented:
The `L.F.Types.Constraints` is getting a bit big; can you put `AxiomEnv` and related in a separate module/file? Thanks!

nikivazou commented:
Shouldn't this have a respective LH pull request that 
- reomoves all the evaluation code from Liquid Haskell 
- uses this fixpoint code instead
- while all the tests still pass?

ranjitjhala commented:
good point...

On Fri, Apr 21, 2017 at 7:50 PM, Niki Vazou <notifications@github.com>
wrote:

> Shouldn't this have a respective LH pull request that
>
>    - reomoves all the evaluation code from Liquid Haskell
>    - uses this fixpoint code instead
>    - while all the tests still pass?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/ucsd-progsys/liquid-fixpoint/pull/305#issuecomment-296203782>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABkuOIeow2HPjZ0OF18AdGVW7_zbIJQ-ks5ryLsqgaJpZM4NDnyr>
> .
>

atondwal commented:
Here's that PR. N.B. Circle won't be able to build or test that PR until this PR is merged
https://github.com/ucsd-progsys/liquidhaskell/pull/993

atondwal commented:
> can you put AxiomEnv and related in a separate module/file?

Hmm, I'm not sure how I should do that `FInfo` contains an `AxiomEnv`, which depends on `SubcId` exported from `L.F.T.Constraints`, so if I made an `L.F.T.AxiomEnv` and put it there, we'd have a circular dependency. We could solve that by moving `FInfo` into its own file too, but that seems like a pretty big change; do we want to do that?

ranjitjhala commented:
ok, lets merge this for now and can fiddle with the AxiomEnv etc. later!

@atondwal I will let you / @nikivazou  merge this in case there are synch issues with LH.

atondwal committed:
Instantiate code cleanup

Reorganized Instantiate a bit and marked string processing hacks for
removal. Additionally, all lines in Instantiate are now <80

atondwal committed:
Merge branch 'develop' into symeval

ranjitjhala commented:
please don't wait for me to merge...

atondwal commented:
I don't have push access

On Apr 28, 2017 18:19, "Ranjit Jhala" <notifications@github.com> wrote:

> please don't wait for me to merge...
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ucsd-progsys/liquid-fixpoint/pull/305#issuecomment-298138078>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAU8CNty5QWYIIOUIZEy2BbT6L1yPt7vks5r0pA6gaJpZM4NDnyr>
> .
>

ranjitjhala commented:
According to github @atondwal has push access...?

atondwal commented:
Oh, I'm so silly; I cloned over https instead of ssh, and was trying to do a `git push origin symeval:develop`, and obviously that didn't work :embarassed:

ranjitjhala commented:
Apparently in such situations one can just edit
the .git/config file and make the `url` field be `git@github.com...`

On Sat, Apr 29, 2017 at 3:30 PM, Anish Tondwalkar <notifications@github.com>
wrote:

> Oh, I'm so silly; I cloned over https instead of ssh, and was trying to do
> a git push origin symeval:develop, and obviously that didn't work
> :embarassed:
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/ucsd-progsys/liquid-fixpoint/pull/305#issuecomment-298159502>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABkuOFko95G6YurEdwviOpY9NqB1TdAYks5r0wpFgaJpZM4NDnyr>
> .
>
